lib/src/widgets/button_widget.dart

```diff
@@ -0,0 +1,340 @@
+
+  ButtonWidget({
+    Key? key,
+    required this.command,
+    required this.state,
+  }) : super(key: key);
+
+  @override
+  _ButtonWidgetState createState() => _ButtonWidgetState();
+}
+
+class _ButtonWidgetState extends State<ButtonWidget> {
+  late final String buttonText;
+  late final int actionType;
+  late final String actionParameter;
+
+  @override
+  void initState() {
+    super.initState();
+    _parseCommand();
+  }
+
+  void _parseCommand() {
+    final buttonExp = RegExp(r'%% Button\((.+)\)');
+    final buttonMatch = buttonExp.firstMatch(widget.command);
+
+    if (buttonMatch != null) {
+      final argsString = buttonMatch.group(1)!;
+      final args = _parseArguments(argsString);
+
+      if (args.isNotEmpty) {
+        buttonText = args.isNotEmpty ? args[0] : defaultButtonText;
+        actionType = args.length > 1 ? int.tryParse(args[1]) ?? 0 : 0;
+        actionParameter = args.length > 2 ? args[2] : defaultFileName;
+      } else {
+        buttonText = defaultButtonText;
+        actionType = 0;
+        actionParameter = defaultFileName;
+      }
+    } else {
+      buttonText = defaultButtonText;
+      actionType = 0;
+      actionParameter = defaultFileName;
+    }
+  }
+
+  /// Collects the user responses from the state map.
+  Map<String, dynamic> _collectData() {
+    // Access the state variables and collect data
+    final Map<String, dynamic> responses = {};
+
+    final _inputValues = widget.state['_inputValues'] as Map<String, String>;
+    final _sliderValues = widget.state['_sliderValues'] as Map<String, double>;
+    final _radioValues = widget.state['_radioValues'] as Map<String, String?>;
+    final _checkboxValues =
+        widget.state['_checkboxValues'] as Map<String, Set<String>>;
+    final _dateValues = widget.state['_dateValues'] as Map<String, DateTime?>;
+    final _dropdownValues =
+        widget.state['_dropdownValues'] as Map<String, String?>;
+
+    // Add slider values
+    _sliderValues.forEach((key, value) {
+      responses[key] = value;
+    });
+
+    // Add radio values
+    _radioValues.forEach((key, value) {
+      responses[key] = value;
+    });
+
+    // Add checkbox values
+    _checkboxValues.forEach((key, value) {
+      // Convert Set to List
+      responses[key] = value.toList();
+    });
+
+    // Add date values
+    _dateValues.forEach((key, value) {
+      if (value != null) {
+        responses[key] =
+            '${value.year}-${value.month.toString().padLeft(2, '0')}-'
+            '${value.day.toString().padLeft(2, '0')}';
+      } else {
+        responses[key] = null;
+      }
+    });
+
+    // Add dropdown values
+    _dropdownValues.forEach((key, value) {
+      responses[key] = value;
+    });
+
+    // Add text input values
+    _inputValues.forEach((key, value) {
+      responses[key] = value;
+    });
+
+    return responses;
+  }
+
+  Future<void> _handleButtonPress() async {
+    final data = _collectData();
+
```

Comment:
The following is quite long and nested. Is there an opportunity to define some local functions and use them here so that we can read the code mode readily.